virttest/qemu_vm.py

```diff
@@ -1897,6 +1898,46 @@ def sort_key(dev):
                               pci_bus=pci_bus)
                 iov += 1
 
+        def get_cid(cid):
+            """ Get an unused guest cid from system """
+            while cid:
+                cid_c = struct.pack('L', cid)
+                try:
+                    fcntl.ioctl(
+                        vsock_fd, arch.VHOST_VSOCK_SET_GUEST_CID, cid_c)
+                except IOError as e:
+                    if e.errno == 98:    # [Errno 98] Address already in use
+                        cid += 1
+                        continue
+                    else:
+                        raise e
+                else:
+                    return cid
+
+        # Add vsock device, cid 0-2 are reserved by system
+        min_cid = 3
+        vsock_path = "/dev/vhost-vsock"
+        try:
+            vsock_fd = open(vsock_path, 'r+')
+        except IOError as e:
+            if e.errno == 2:  # [Errno 2] No such file or directory
+                logging.info("vsock module was not loaded, loading it...")
+                process.system("modprobe vsock && modprobe vhost_vsock")
+                vsock_fd = open(vsock_path, 'r+')
+            else:
+                raise e
```

Comment:
Suggest the following approach:
```python
from avocado.utils import linux_module

min_cid = 3
vsock_path = "/dev/vhost-vsock"
vsock_fd = None
for vsock in params.objects('vsocks'):
    if vsock_fd is None:
        if not os.path.exists(vsock_path):
            if not linux_module.load_module('vhost_vsock'):
                test.error("...")
        vsock_fd = os.open(vsock_path, ...)
    ...
if vsock_fd is not None:
    os.close(vsock_fd)
```

Reply:
@luckyh Yes, but I would like to load the module and get vsock_fd out of the for loop, it only needs to be executed once.

Reply:
Well, if user does not use vsock devices, `/dev/vhost-vsock` does not have to be opened, hence a module missing issue would not be raised in unsupported platforms.

Reply:
@luckyh You are right. Updating.